ngx_http_sxg_filter_module.c

```diff
@@ -303,38 +333,123 @@ static ngx_str_t extract_angled_url(char* str, size_t len) {
+          ngx_memcpy(new_preload->url.data, url.data, url.len);
+          while (new_preload->url.data[new_preload->url.len - 1] == ' ') {
+            new_preload->url.len--;
+          }
+          if (as.data != NULL) {
+            new_preload->as = as;
+          }
+        }
+        found = true;
+      }
+      if (param_is_as(param, param_tail, (const char**)&as.data, &as.len)) {
+        if (found && new_preload != NULL) {
+          new_preload->as = as;
+          ngx_log_error(
+              NGX_LOG_DEBUG, r->connection->log, 0,
+              "nginx-sxg-module: found as statement in link preload header: %V",
+              &new_preload->as);
+        }
       }
       param += param_tail + 1;
     }
+    if (!found) {
+      if (non_preload_headers == NULL) {
+        if (non_preload_headers_len > 0) {
+          ++non_preload_headers_len;
+        }
+        non_preload_headers_len += tail;
+      } else {
+        if (non_preload_headers->len > 0) {
+          ngx_memcpy(non_preload_headers->data + non_preload_headers_len, ",",
+                     1);
+          ++non_preload_headers_len;
+        }
+        ngx_memcpy(non_preload_headers->data + non_preload_headers_len, str,
+                   tail);
+        non_preload_headers_len += tail;
+      }
+    }
     str += tail + 1;
   }
+
+  return non_preload_headers_len;
 }
 
-static ngx_array_t* get_preload_list(ngx_str_t* link, ngx_http_request_t* req) {
-  ngx_array_t* const urls = ngx_array_create(req->pool, 1, sizeof(ngx_str_t));
-  extract_preaload_url_list(link, urls, req);
+static ngx_array_t* get_preload_list(ngx_str_t* link,
+                                     ngx_str_t* non_preload_headers,
+                                     ngx_http_request_t* req) {
+  ngx_array_t* const urls =
+      ngx_array_create(req->pool, 1, sizeof(ngx_subresource_t));
+  size_t non_preload_headers_len =
+      extract_preload_url_list(link, NULL, NULL, req);
+  non_preload_headers->data = ngx_palloc(req->pool, non_preload_headers_len);
+  non_preload_headers->len = non_preload_headers_len;
+  extract_preload_url_list(link, urls, non_preload_headers, req);
   return urls;
 }
 
+static bool set_str(ngx_pool_t* pool, ngx_str_t* dst, const char* src) {
+  const size_t len = strlen(src);
+  dst->data = ngx_palloc(pool, len);
+  if (dst->data == NULL) {
+    return false;
+  }
+  dst->len = len;
+  memcpy(dst->data, src, len);
+  return true;
+}
+
+static bool set_accept_headers(ngx_http_request_t* req,
+                               ngx_subresource_t* target, const char* accept) {
+  const static char* kAccept = "accept";
+  ngx_list_part_t* part = &req->headers_in.headers.part;
+  ngx_table_elt_t* v = part->elts;
+
+  for (int i = 0; i < part->nelts; i++) {
+    if (strncasecmp(kAccept, (const char*)v[i].key.data, strlen(kAccept)) ==
+            0 &&
+        !set_str(req->pool, &v[i].value, accept)) {
+      return false;
+    }
+  }
+  return true;
```

Comment:
Do we need to append an accept header if none is found, or will that never happen?